module_4/src/com/epam/aggregationandcomposition/entity/car/Car.java

```diff
@@ -0,0 +1,128 @@
+package com.epam.aggregationandcomposition.entity.car;
+
+import com.epam.aggregationandcomposition.exceptions.CarLogicalException;
+import com.epam.aggregationandcomposition.exceptions.CarTechnicalException;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class Car {
+
+    private static final float FULL_TANK = 1.0f;
+    private static final float FUEL_AMOUNT_TO_ONE_RIDE = 0.2f;
+    private static final byte NUMBER_OF_WHEELS = 4;
+
+    String carBrand;
+    Engine engine;
+    Wheel[] wheels;
+    float fuelLevel = 1.0f;
+
+    public Car(String carBrand, Engine engine, Wheel[] wheels) throws CarLogicalException {
+        if (carBrand != null && engine != null &&  wheels != null) {
+            this.carBrand = carBrand;
+            this.engine = engine;
+            this.wheels = wheels;
+        } else {
+            throw new CarLogicalException("Wrong given parameters. Can't create a car!");
+        }
+    }
+
+    public float go() throws CarTechnicalException {
```

Comment:
I think you can make it just void
```suggestion
    public void go() throws CarTechnicalException {
```